src/configuration.js

```diff
@@ -365,6 +362,10 @@ Configuration.prototype._checkLocalProjectId = function(cb) {
     } else {
       cb(new Error('config.projectId must be a string or number'), null);
     }
+  } else if (isString(env.GCLOUD_PROJECT) && !isEmpty(env.GCLOUD_PROJECT)) {
```

Comment:
I think `process.env.x` for any `x` is either `undefined` or a string.  Perhaps a slightly different test could be done to make this explicit or a comment added to point this out.